onnx/shape_inference/implementation.cc

```diff
@@ -211,23 +209,24 @@ static void InferShapesImpl(
     const auto schema =
         schema_registry->GetSchema(n.op_type(), domain_version, n.domain());
     InferenceContextImpl ctx(
-        n, valueTypesByName, inputDataByName, &graphInferenceContext);
+        n, valueTypesByName, inputDataByName, &graphInferenceContext);    
     if (!schema) {
       continue;
     } else if (schema->has_type_and_shape_inference_function()){
       try {
         schema->GetTypeAndShapeInferenceFunction()(ctx);
       } catch (const ONNX_NAMESPACE::InferenceError& ex) {
-        (void)ex;
         // Continue with inference for remaining nodes
+        inference_errors.push_back(std::string(ex.what()));
         continue;
       }
     } else if (schema->HasFunction()) {
       try {
         InferShapeForFunctionNode(
           schema->GetFunction(), schema_registry, ctx);
       } catch (const ONNX_NAMESPACE::InferenceError& function_ex) {
-        (void)function_ex;
+        // Continue with inference for remaining nodes
+        inference_errors.push_back(std::string(function_ex.what()));
```

Comment:
I think we should replace the checks in line 243 and 248 by "if (inferredType.value_case() == VALUE_NOT_SET)".